os/os.proto

```diff
@@ -0,0 +1,321 @@
+  // the RPC MUST immediately abort with Error->type->UNEXPECTED_SWITCHOVER.
+  //
+  // Scenario 3 - When both Supervisors already have the OS package, regardless
+  //              of the value in Start.standby_supervisor:
+  //
+  //         Client :--------------|--------------> Target
+  //              TransferRequest -->
+  //                              <-- [Validated|InstallError]
+  //
+  //
+  // Scenario 4 - When one of the Supervisors already has the OS package but the
+  //              other Supervisor is the target of the Install:
+  //
+  //         Client :--------------|--------------> Target
+  //              TransferRequest -->
+  //                              <-- [SyncProgress|InstallError]
+  //                              ...
+  //                              <-- [Validated|InstallError]
+  //
+  //
+  // Scenario 5 - When neither of the two Supervisors has the OS package:
+  //
+  //         Client :--------------|--------------> Target
+  //              TransferRequest -->
+  //                              <-- [TransferReady|InstallError]
+  //            transfer_content  -->
+  //                              ...
+  //                              <-- [TransferProgress|InstallError]
+  //                              ...
+  //                  TransferEnd -->
+  //                              <-- [Validated|InstallError]
+  //
+  rpc Install(stream InstallRequest) returns (stream InstallResponse);
+
+  // Activate sets the requested OS version as the version which is used at the
+  // next reboot, and reboots the Target. When booting the requested OS version
+  // fails, the Target recovers by booting the previously running OS package.
+  rpc Activate(ActivateRequest) returns (ActivateResponse);
+
+  // Verify checks the running OS version. This RPC may be called multiple times
+  // while the Target boots, until successful.
+  rpc Verify(VerifyRequest) returns (VerifyResponse);
+}
+
+message InstallRequest {
+  oneof request {
+    TransferRequest transfer_request = 1;
+    bytes transfer_content = 2;
+    TransferEnd transfer_end = 3;
+  }
+}
+
+message TransferRequest {
+  // State the desired OS package version to transfer to the Target. If the
+  // Target already has the OS package version it will reply with
+  // InstallResponse->Validated. In the case that the target is a
+  // single Supervisor device, or the partner Supervisor does not have the OS
+  // image specified, it will respond with InstallResponse->TransferReady. In
+  // this case, the client MUST subsequently transfer the image. In the case
+  // that the image is available on the peer Supervisor of a dual Supervisor
+  // system, it will respond with InstallResponse->SyncProgress. In this,
+  // latter, case - the client does not need to transfer the OS image. This
+  // value can also be set empty, in which case the OS package is forced
+  // transferred to the Target. The Target MUST never validate that this value
```

Comment:
When the version is empty, the target generates a version/name for the package?

Reply:
The version string is always a well known OS version string builtin in the OS package and provided by the vendor. Will update text. thanks!